# Board Meeting Report
----------------
##### By:
Alejandro Velasquez 

David Rodriguez


- To replicate data, stakeholder should have propper env file

In [1]:
# Imports
import itertools
import warnings
warnings.filterwarnings("ignore")
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.dates as mdates
import numpy as np
import pandas as pd
import math

import env
import os
import wrangle

from sklearn import metrics
from random import randint
from matplotlib import style

from sklearn.cluster import DBSCAN
from sklearn.preprocessing import MinMaxScaler

In [2]:
df = wrangle.curriculum_logs_data()

In [3]:
df.head()

date      time                                 path  user_id  \
0  2018-01-26  09:55:03                                    /        1   
1  2018-01-26  09:56:02                              java-ii        1   
2  2018-01-26  09:56:05  java-ii/object-oriented-programming        1   
3  2018-01-26  09:56:06   slides/object_oriented_programming        1   
4  2018-01-26  09:56:24            javascript-i/conditionals        2   

   cohort_id            ip    id     name     slack  start_date    end_date  \
0        8.0  97.105.19.61   8.0  Hampton  #hampton  2015-09-22  2016-02-06   
1        8.0  97.105.19.61   8.0  Hampton  #hampton  2015-09-22  2016-02-06   
2        8.0  97.105.19.61   8.0  Hampton  #hampton  2015-09-22  2016-02-06   
3        8.0  97.105.19.61   8.0  Hampton  #hampton  2015-09-22  2016-02-06   
4       22.0  97.105.19.61  22.0    Teddy    #teddy  2018-01-08  2018-05-17   

            created_at           updated_at  deleted_at  program_id  
0  2016-06-14 19:52:26  2016-06-14 19:52:26         NaN         1.0  
1  2016-06-14 19:52:26  2016-06-14 19:52:26         NaN         1.0  
2  2016-06-14 19:52:26  2016-06-14 19:52:26         NaN         1.0  
3  2016-06-14 19:52:26  2016-06-14 19:52:26         NaN         1.0  
4  2018-01-08 13:59:10  2018-01-08 13:59:10         NaN         2.0

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 900223 entries, 0 to 900222
Data columns (total 15 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   date        900223 non-null  object 
 1   time        900223 non-null  object 
 2   path        900222 non-null  object 
 3   user_id     900223 non-null  int64  
 4   cohort_id   847330 non-null  float64
 5   ip          900223 non-null  object 
 6   id          847330 non-null  float64
 7   name        847330 non-null  object 
 8   slack       847330 non-null  object 
 9   start_date  847330 non-null  object 
 10  end_date    847330 non-null  object 
 11  created_at  847330 non-null  object 
 12  updated_at  847330 non-null  object 
 13  deleted_at  0 non-null       float64
 14  program_id  847330 non-null  float64
dtypes: float64(4), int64(1), object(10)
memory usage: 109.9+ MB


In [5]:
df = wrangle.wrangle_curriculum_logs_data(df)

In [6]:
df.head(2)

date      time endpoint  user_id  cohort_id     source_ip  \
date                                                                         
2018-01-26 2018-01-26  09:55:03        /        1        8.0  97.105.19.61   
2018-01-26 2018-01-26  09:56:02  java-ii        1        8.0  97.105.19.61   

           cohort_name start_date   end_date          created_at  \
date                                                               
2018-01-26     Hampton 2015-09-22 2016-02-06 2016-06-14 19:52:26   
2018-01-26     Hampton 2015-09-22 2016-02-06 2016-06-14 19:52:26   

                     updated_at  program_id           date_time  \
date                                                              
2018-01-26  2016-06-14 19:52:26         1.0 2018-01-26 09:55:03   
2018-01-26  2016-06-14 19:52:26         1.0 2018-01-26 09:56:02   

                              program_name           course  
date                                                         
2018-01-26  PHP Full Stack Web Development  Web Development  
2018-01-26  PHP Full Stack Web Development  Web Development

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 763298 entries, 2018-01-26 to 2021-04-21
Data columns (total 15 columns):
 #   Column        Non-Null Count   Dtype         
---  ------        --------------   -----         
 0   date          763298 non-null  datetime64[ns]
 1   time          763298 non-null  object        
 2   endpoint      763298 non-null  object        
 3   user_id       763298 non-null  int64         
 4   cohort_id     763298 non-null  float64       
 5   source_ip     763298 non-null  object        
 6   cohort_name   763298 non-null  object        
 7   start_date    763298 non-null  datetime64[ns]
 8   end_date      763298 non-null  datetime64[ns]
 9   created_at    763298 non-null  datetime64[ns]
 10  updated_at    763298 non-null  object        
 11  program_id    763298 non-null  float64       
 12  date_time     763298 non-null  datetime64[ns]
 13  program_name  763298 non-null  object        
 14  course        763298 non-null  object        
dtypes

## Takeaways
- Data was taken from Codeup 'curriculum_logs' data base
- 900223 records where important
- After wrangleing the data only 763298 will be used for report